### DISCUSSION TITLE
[Question]License

### DISCUSSION THREAD

**Discussion by yunle-wang** [luc-github/ESP32SSDP#37] [Q&A] [answered]
In LICENSE file and project indicated this is GPL-3 project, however in source code, there is MIT in place.
I want to double confirm which LIC is for this project? Thank you

[a comment by an automated account is not shown]

**Comment by luc-github** [comment]
original code is mit but my version is gpl3 I will update header

**Comment by f34rdotcom** [comment]
my 2c. I can't use GPL3. It is too restrictive for any commercial use.

**Comment by luc-github** [comment, marked as answer]
according https://softwareengineering.stackexchange.com/questions/105912/can-you-change-code-distributed-under-the-mit-license-and-re-distribute-it-unde
I should keep MIT and add GPL3 - I checked long time ago but forget it
keep MIT header and adding gpl3 is actually correct

**Comment by luc-github** [comment]
@f34rdotcom sorry to read that but all code I do / share should be available if used

**Reply by yunle-wang** [reply]
I have the same concern that any code linked with this project have to open source. This is too restrictive. 
Unless we use API to create a tool, using system call in our code. In that way, we can protect most of commercial code.

**Reply by f34rdotcom** [reply]
No worries its your code :c) maybe I should have our legal check out GPLv3 again looks like some questions are settled that were very unclear back then. I still prefer MIT but if GPLv3 will allow for managed services I am good but last I checked legal said it specifically had provisions preventing it unless used with no changes. Currently all NuTech projects are MIT or Apache except for ser2sock because I specifically did not want anyone to use it for commercial use unless it was unchanged. I may change my mind someday on that one but for now it is GPLv3 for that reason.

**Reply by f34rdotcom** [reply]
> I have the same concern that any code linked with this project have to open source. This is too restrictive. Unless we use API to create a tool, using system call in our code. In that way, we can protect most of commercial code.

As far as I know and I am no expert GPLv3 does not let you use API unless you also share your use of the code. API use it considered to be directly connected to the code so no different than linking its as a binary static.
